### PR TITLE
Use content-tag, which fixes *all the bugs*

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Learned some new cli-fu today:
```
❯ git diff --shortstat HEAD~1 -- . ':(exclude)yarn.lock' 
 33 files changed, 41 insertions(+), 2367 deletions(-)
```
the actual diff! 

# Breaking changes

all consumers that were importing from ember-template-imports will no longer be able to.
folks should typically want to use `content-tag` instead of `ember-template-imports` after this change.

# ~~Blockers~~

- https://github.com/babel/ember-cli-babel/pull/502
- https://github.com/embroider-build/content-tag/pull/24

----------

Extra bonus -- most of the code from the original implementation is gone now :tada: yay deleting code!

(most of the additions are lockfile, btw)
((also most of the removals, but also 98% of the old code is gone, too))

This is a breaking change, however:
- no more `hbs` support
- all downstream dependencies importing from ember-template-imports will need to not import from ember-template-imports.

Related, 
- uses https://github.com/emberjs/babel-plugin-ember-template-compilation/, which also has min ember version of 3.27
- uses https://github.com/embroider-build/content-tag/ for the actual parsing
- removes unused dependencies
  - which also had me remove unused type declarations packages
  - which also had me remove unused type declaration files
- removes every file that would no longer be used
  - this means `tsc` is no longer needed, so I removed that from the package.json scripts
- removes every reference to `build` and `jest` 

Not related,
- I disabled the lint `node/no-missing-require` -- because it's buggy. and eslint-plugin-node hasn't been maintained for quite some time -- long term solution here is to switch to esilnt-plugin-n
- I also disabled the lint `node/no-unpublished-require` -- because it doesn't understand how peer dependencies work.

<details><summary>resolved issues encountered while working on this PR</summary>

What I need help with,
- ~~why did I need to add `@babel/plugin-transform-class-static-block`? browsers support this (I got a babel error about needing to add this)~~
   - RESOLUTION: per https://github.com/ember-template-imports/ember-template-imports/pull/182, I need to make the babel plugins peers -- it's intended that the plugins are needed due to the decorator transform our community is using.
- looks like the test suite was maybe never passing against ember v5? -- it may take me a bit to figure out what's going on there
  UPDATE: I have things working on stackblitz -- so I don't yet know what's going on with C.I.: 
  image

</details>

Supersedes https://github.com/ember-template-imports/ember-template-imports/pull/182

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
Failures exist over here: https://github.com/ember-template-imports/ember-template-imports/pull/190, so it has nothing to do with content-tag

**Comment by chriskrycho** [comment]
What does this mean?

> all downstream dependencies importing from ember-template-imports will need to not import from ember-template-imports.

**Comment by NullVoxPopuli** [comment]
> > all downstream dependencies importing from ember-template-imports will need to not import from ember-template-imports.
> 
> What does this mean?

 eslint-plugin-ember and ember-template-lint will need to remove their (peer)dependencies on ember-template-imports, because the `util` and `GLIMMER_TEMPLATE` constant have been removed. 
 
 prettier-plugin-ember-template-tag should be fine because it declares its own hard dependency on `ember-template-imports`.
 
There are paths forward for each of these, and @patricklx has been doing awesome work making _legit_ lint tooling and I don't know what or if there is a story for combining efforts with content-tag, but the benefits, impact, and desire for getting the tools off of ember-template-imports is described here: https://github.com/ember-template-imports/ember-template-imports/issues/143

File: ember-addon-main.js [hunk 1]
```diff
@@ -17,25 +17,30 @@ module.exports = {
...
+      '@babel/plugin-transform-class-static-block'
+    );
 
-    if (!hasPlugin(includer, pluginPath)) {
-      addPlugin(includer, pluginPath);
-    }
+    addPlugin(includer, staticBlocks);
 
-    // Used in ember-cli-htmlbars to get the location of templateCompiler without traversing this.addons (https://github.com/ember-cli/ember-cli-htmlbars/blob/6860beed9a357d5e948abd09754e8a978fed1320/lib/ember-addon-main.js#L264)
+    let pluginPath = require.resolve('babel-plugin-ember-template-compilation');
```
Comment (ef4) [inline ember-addon-main.js:26]: I don't think any of this should be here. All ember apps already have this plugin. 

They get it via ember-cli-htmlbars or embroider. 

I think we should document that ember-template-imports has a minimum supported version of ember-cli-htmlbars and/or embroider and leave it at that.

Comment (ef4) [reply ember-addon-main.js:26]: And we'd update ember-cli-htmlbars and embroider so there are releases that require the new-enough babel-plugin-ember-template-compilation.

Also, those updates would probably need to add the static-block babel plugin, in the same place where they currently add the other class feature plugins that are forced by legacy decorators.

Comment (NullVoxPopuli) [reply ember-addon-main.js]: sounds good, I made "a" change here in this PR, which I expect to fail until we update ember-cli-htmlbars and embroider

File: package.json [hunk 2]
```diff
@@ -16,29 +16,24 @@
...
     "start": "ember serve",
     "test": "npm-run-all lint test:*",
-    "test:jest": "jest",
     "test:ember": "ember test",
     "test:ember-compatibility": "ember try:each"
   },
   "dependencies": {
-    "babel-import-util": "^0.2.0",
+    "@babel/plugin-transform-class-static-block": "^7.22.5",
+    "babel-plugin-ember-template-compilation": "^2.2.0",
```
Comment (ef4) [inline package.json:32]: These don't belong here.

Comment (NullVoxPopuli) [reply package.json]: thanks!

Comment (NullVoxPopuli) [reply package.json]: it seems that ember-cli-babel is not automatically adding `@babel/plugin-proposal-class-static-block` now

Comment (NullVoxPopuli) [reply package.json]: looks like the lockfile's browserslist was way out of date :sweat_smile: 

```
Latest version:     1.0.30001538
Installed versions: 1.0.30001425, 1.0.30001538
```

**Comment by NullVoxPopuli** [comment]
There's a bit of bitrot, so I'm going to open some other PRs to try to make this easier.
- ember-cli-babel v8 is needed
- ember-cli-htmlbars needs upgraded
- ember-qunit needs upgraded (to not use ember-cli-typescript)
  (this'll require some other upgrades as well, like ember-cli)

My hope is that I can avoid waiting for https://github.com/typed-ember/ember-cli-typescript/pull/1569 as well, but timing might work out where it's no big deal.

**Comment by NullVoxPopuli** [comment]
Current situation:
```
WARNING: ember-compatibility-helpers: You are using an unsupported ember-cli-babel version, compatibility helper tranforms will not be included automatically
WARNING: ember-cli-typescript requires ember-cli-babel ^7.7.3, but you have version 8.1.0 installed; your TypeScript files may not be transpiled correctly.
WARNING: ember-compatibility-helpers: You are using an unsupported ember-cli-babel version, compatibility helper tranforms will not be included automatically
WARNING: ember-cli-typescript requires ember-cli-babel ^7.1.0, but you have version 8.1.0 installed; your TypeScript files may not be transpiled correctly.
Build Error (Analyzer)

don't know how to setup a parser for Babel version 8 (used by ember-qunit)


Stack Trace and Error Report: /tmp/error.dump.eb740d675201b890f76b3a5246f1d90c.log
```

**Comment by NullVoxPopuli** [comment]
Running in to:
```
Error: Could not find module `ember-compatibility-helpers` imported from `@glimmer/component/index`
                at missingModule (http://localhost:7357/assets/vendor.js:266:11)
                at findModule (http://localhost:7357/assets/vendor.js:277:7)
                at Module.findDeps (http://localhost:7357/assets/vendor.js:187:24)
                at findModule (http://localhost:7357/assets/vendor.js:281:11)
                at Module.findDeps (http://localhost:7357/assets/vendor.js:187:24)
```

I remember ember-compatibility-helpers having issues throughout the years. 

The modern way do to what it wants to do is embroider/macros.
which.. would require a bunch more changes to other packages -- so hoping that's not needed

**Comment by NullVoxPopuli** [comment]
We need to merge / release:
- https://github.com/ember-cli/ember-compatibility-helpers/pull/57
- and https://github.com/typed-ember/ember-cli-typescript/pull/1569 (tho, I've had a really hard time getting this repo green, which is a smidge frustrating)

Gonna switch this project to pnpm in another PR tho, because yarn can't handle the state of our dependencies right now (locally, I have tests passing, but in yarn, I have the above ember-compatibility-helpers issue, which is typically due to incorrect duplicate dependencies in the dep graph)

**Comment by NullVoxPopuli** [comment]
this PR now builds on top of: https://github.com/ember-template-imports/ember-template-imports/pull/203

**Comment by patricklx** [comment]
It says it fixes all the bugs. But didn't close any GitHub issues :)

**Comment by NullVoxPopuli** [comment]
Oops. Thanks!

**Pull request merged**